File: Backend/BankingPaymentsApp_API/BankingPaymentsApp_API/Models/SalaryDisbursement.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BankingPaymentsApp_API.Models
{
    public class SalaryDisbursement
    {
        [Key]
        public int SalaryDisbursementId { get; set; }
        [ForeignKey("ClientUser")]
        public int ClientId {  get; set; }
        public virtual ClientUser? ClientUser { get; set; }
        [Required(ErrorMessage = "Total Amount is Required!")]
        [DataType(DataType.Currency)]
        public decimal TotalAmount { get; set; }
        [DataType(DataType.DateTime)]
        public DateTime DisbursementDate { get; set; } = DateTime.UtcNow;
        [ForeignKey("DisbursementStatus")]
        public int DisbursementStatusId { get; set; } = 3;
        public virtual PaymentStatus? DisbursementStatus { get; set; }
        [Required(ErrorMessage = "AllEmployee check is Mandatory")]
        public bool AllEmployees { get; set; } = true;
        public virtual ICollection<Employee>? Employees { get; set; } = new List<Employee>();
        public virtual ICollection<SalaryDisbursementDetails>? DisbursementDetails { get; set; } = new List<SalaryDisbursementDetails>();

    }
}

File: frontend/BankingPaymentsApp_Frontend/src/app/Models/SalaryDisbursement.ts
import { ClientUser } from "./ClientUser";
import { Employee } from "./Employee";
import { PaymentStatus } from "./PaymentStatus";
import { SalaryDisbursementDetails } from "./SalaryDisbursementDetails";

export interface SalaryDisbursement {
  salaryDisbursementId: number;
  clientId: number;
  clientUser?: ClientUser;
  totalAmount: number; 
  disbursementDate: string; 
  disbursementStatusId: number;
  disbursementStatus?: PaymentStatus;
  allEmployees: boolean;
  employees?: Array<Employee>;
  disbursementDetails?: Array<SalaryDisbursementDetails>;
}